File: java/src/test/java/problems201_300/BinaryTreePathsTest.java
package problems201_300;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class BinaryTreePathsTest extends SampledTest {
    @Test
    public void test() {
        for (BinaryTreePaths solution : BinaryTreePaths.values()) {
            for (BinaryTreePathsSample sample : getSamples(BinaryTreePathsSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.binaryTreePaths(TreeNode.of(sample.input.root)))
                    .containsExactlyElementsIn(sample.output)
                    .inOrder();
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/BinaryTreePathsTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class BinaryTreePathsTest : SampledTest() {
    @Test
    fun test() =
        BinaryTreePaths.entries.forEach { solution ->
            getSamples<Array<BinaryTreePathsSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.binaryTreePaths(TreeNode.of(*input.root)))
                    .containsExactlyElementsIn(output)
                    .inOrder()
            }
        }
}
